### ISSUE TITLE
Core is open source, (enterprise) features remain closed source

### ISSUE THREAD

**Issue by jayfk** [pybee/paying-the-piper#9]
Nginx Inc. the company behind nginx is using a model like this. Nginx is free and open source, but they offer a pro version called Nginx Plus to fund development. The Plus version is subscription based and targeted to enterprise customers.

**Comment by freakboy3742** [comment]
A problem with this approach is that the enterprise features don't get the engineering benefit of being open source - more eyeballs on bugs, etc.

A variation on this theme is to hold back features from the open source version until a certain time passes. Aladdin Ghostscript did this for a while - if you wanted the latest, greatest code, pay us; otherwise, wait 6-12 months and get it in the open source pool.

**Comment by lukego** [comment]
This is an exclusive model: it helps one company make money on the project but not others.

**Comment by thbar** [comment]
An instance of this model is [Sidekiq](http://sidekiq.org/) by @mperham (maybe Mike you'll want to share more about your experience here?). I think Mike's Inspektor Pro is open-source too, by the way (a different model).

**Comment by nanuxbe** [comment]
+1 on the main idea

**Comment by mperham** [comment]
Yep, I sell closed-source extensions to Sidekiq's open core as my full-time job.  The source is actually available (it is normally packaged Ruby code) but not without payment.

**Comment by wolftune** [comment]
This is not a model for software freedom, it's a model for half-assed freemium software freedom. At _best_, it's like GitLab where the extra features are really not that significant a set, but still **this is a _symptom_ of the funding problem, not a solution to it**.

**Comment by cvrebert** [comment]
There's also the issue of what happens if someone from the community tries to contribute their own implementation of an Enterprise™ feature to the open-source version.
